File: expence_app/lib/enums/transaction_category.dart
import 'package:flutter/material.dart';

// ignore_for_file: constant_identifier_names

enum TransactionCategory {
  miscellaneous,
  transportation,
  services,
  rent_assets,
  purchases,
  students,
  support,
}

extension TransactionCategoryExtension on TransactionCategory {
  String get description {
    switch (this) {
      case TransactionCategory.miscellaneous:
        return 'نثريات';
      case TransactionCategory.transportation:
        return 'مواصلات';
      case TransactionCategory.services:
        return 'خدمات';
      case TransactionCategory.rent_assets:
        return 'أصول إيجار';
      case TransactionCategory.purchases:
        return 'مشتريات';
      case TransactionCategory.students:
        return 'طلاب';
      case TransactionCategory.support:
        return 'دعم';
    }
  }

  IconData get icon {
    switch (this) {
      case TransactionCategory.miscellaneous:
        return Icons.receipt_long; // أيقونة النثريات
      case TransactionCategory.transportation:
        return Icons.directions_car; // أيقونة المواصلات
      case TransactionCategory.services:
        return Icons.miscellaneous_services_sharp; // أيقونة الخدمات
      case TransactionCategory.rent_assets:
        return Icons.apartment  ; // أيقونة أصول الإيجار
      case TransactionCategory.purchases:
        return Icons.shopping_cart; // أيقونة المشتريات
      case TransactionCategory.students:
        return Icons.people_alt; // أيقونة الطلاب
      case TransactionCategory.support:
        return Icons.volunteer_activism; // أيقونة الدعم
    }
  }
}

File: backend/app/Enums/TransactionCategory.php
<?php

namespace App\Enums;

use BenSampo\Enum\Enum;

final class TransactionCategory extends Enum
{
    const MISCELLANEOUS = 'miscellaneous'; // تأكد من وجود هذه القيمة
    const TRANSPORTATION = 'transportation';
    const SERVICES = 'services';
    const RENT_ASSETS = 'rent_assets';
    const PURCHASES = 'purchases';
    const STUDENTS = 'students';
    const SUPPORT = 'support';

    public static function getDescription($value): string
    {
        switch ($value) {
            case self::MISCELLANEOUS:
                return 'نثريات';
            case self::TRANSPORTATION:
                return 'مواصلات';
            case self::SERVICES:
                return 'خدمات';
            case self::RENT_ASSETS:
                return 'أصول إيجار';
            case self::PURCHASES:
                return 'مشتريات';
            case self::STUDENTS:
                return 'طلاب';
            case self::SUPPORT:
                return 'دعم';
            default:
                return self::getKey($value);
        }
    }
}
